src/Microsoft.DocAsCode.Build.Common/MarkdownReader.cs

```diff
@@ -33,50 +37,118 @@ public static List<OverwriteDocumentModel> ReadMarkdownAsOverwrite(string baseDi
+                return parts;
+            }
+            catch (Exception ex)
+            {
+                System.Diagnostics.Debug.Fail("Markup failed!");
+                Logger.LogWarning($"Markup failed:{Environment.NewLine}  Markdown: {markdown}{Environment.NewLine}  Details:{ex.ToString()}");
+                return Enumerable.Empty<YamlHtmlPart>();
+            }
+        }
+
+        private static MatchDetail Select(YamlHtmlPart part)
+        {
+            if (part == null)
+            {
+                return null;
+            }
+
+            var properties = part.YamlHeader;
+            string checkPropertyMessage;
+            var checkPropertyStatus = CheckRequiredProperties(properties, RequiredProperties, out checkPropertyMessage);
+            if (!checkPropertyStatus)
+            {
+                throw new InvalidDataException(checkPropertyMessage);
+            }
+
+            var overriden = RemoveRequiredProperties(properties, RequiredProperties);
+
+            return new MatchDetail
+            {
+                Id = properties[Constants.PropertyName.Uid].ToString(),
+                Properties = overriden
+            };
+        }
+
+        private static readonly List<string> RequiredProperties = new List<string> { Constants.PropertyName.Uid };
+
+        private static Dictionary<string, object> RemoveRequiredProperties(ImmutableDictionary<string, object> properties, IEnumerable<string> requiredProperties)
+        {
+            if (properties == null) return null;
+
+            var overridenProperties = new Dictionary<string, object>(properties);
+            foreach (var requiredProperty in requiredProperties)
+            {
+                if (requiredProperty != null) overridenProperties.Remove(requiredProperty);
             }
+
+            return overridenProperties;
         }
 
-        private static IEnumerable<int> GetLineIndex(string content)
+        private static bool CheckRequiredProperties(ImmutableDictionary<string, object> properties, IEnumerable<string> requiredKeys, out string message)
         {
-            var index = 0;
-            while (index >= 0)
+            var requiredKeyExistence = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+            foreach (var requiredKey in requiredKeys)
             {
-                yield return index;
-                index = content.IndexOf('\n', index + 1);
+                bool current;
+                if (!requiredKeyExistence.TryGetValue(requiredKey, out current))
+                {
+                    requiredKeyExistence.Add(requiredKey, false);
+                }
+            }
+
+            foreach (var property in properties)
+            {
+                if (requiredKeyExistence.ContainsKey(property.Key))
+                {
+                    requiredKeyExistence[property.Key] = true;
+                }
             }
+
+            var notExistsKeys = requiredKeyExistence.Where(s => !s.Value);
```

Comment:
Can we use `requiredKeys.Where(k => !properties.Contains(k, StringComparer.OrdinalIgnoreCase)); ` ?